migration/migrator.go

```diff
@@ -215,28 +215,29 @@ func (m *StateMigrator) migrateStorage(
 		return common.Hash{}, err
 	}
 
-	var mu sync.Mutex
+	nodeIt, err := zkt.NodeIterator(nil)
+	if err != nil {
+		return common.Hash{}, fmt.Errorf("failed to open node iterator (root: %s): %w", zkt.Hash(), err)
+	}
+	iter := trie.NewIterator(nodeIt)
 	var slots atomic.Uint64
-	err = hashRangeIterator(m.ctx, zkt, NumProcessStorage, func(key, value []byte) error {
```

Comment:
I think by removing `hashRangeIterator` which contains checking the `m.ctx.Done()` signal, if `m.ctx.Done()` signal arrives while executing `migrateStorage`, `migrateAccount` will wait until `migrateStorage` execution completed. How about adding L131-L142 goroutine in `migrateStorage` function also?

Reply:
Oh, do you mean adding logs for the migrated storage number in the goroutine, or halting the iteration process immediately when m.ctx.Done() is received, by using a goroutine?

Reply:
I meant that if `m.ctx.Done()` is received, `migrateAccount` and `migrateStorage` should also return immediately. You are right, the current goroutine returns the logging only. Then how about adding below code block to below L179?
```go
		select {
		case <-m.ctx.Done():
			return m.ctx.Err()
		default:
		}
```

Reply:
Good idea. I’ve applied it.